SignalMessaging/attachments/SharingThreadPickerViewController.m

```diff
@@ -158,15 +200,43 @@ - (void)attachmentApproval:(AttachmentApprovalViewController *)attachmentApprova
     [self cancelShareExperience];
 }
 
+#pragma mark - MessageApprovalViewControllerDelegate
+
+- (void)messageApproval:(MessageApprovalViewController *)approvalViewController
+      didApproveMessage:(NSString *)messageText
+{
+    OWSAssert(messageText.length > 0);
+
+    [ThreadUtil addThreadToProfileWhitelistIfEmptyContactThread:self.thread];
+    [self tryToSendMessageWithBlock:^(SendCompletionBlock sendCompletion) {
+        __block TSOutgoingMessage *outgoingMessage = nil;
+        outgoingMessage = [ThreadUtil sendMessageWithText:messageText
+            inThread:self.thread
+            messageSender:self.messageSender
+            success:^{
+                sendCompletion(nil, outgoingMessage);
+            }
+            failure:^(NSError *_Nonnull error) {
+                sendCompletion(error, outgoingMessage);
+            }];
+    }
+                 fromViewController:approvalViewController];
+}
+
+- (void)messageApprovalDidCancel:(MessageApprovalViewController *)approvalViewController
+{
+    [self cancelShareExperience];
+}
+
 #pragma mark - Helpers
+typedef void (^SendMessageBlock)(SendCompletionBlock completion);
```

Comment:
this is redundant with the declaration at the top of the file

Reply:
Removed.